### ISSUE TITLE
Simple example

### ISSUE THREAD

**Issue by lucapaganotti** [SOCI/soci#769]
Hi, new to soci, I'm trying to educate myself trying with simple examples.

So I created a very simple table with two fields: an int not null and a varchar(50) null in a sql server instance.
I wrote then a simple c++ project linking soci libraries, setup an ODBC session and ask for data.

I'm working on a debian buster box which delivers soci 3.2 packages. All the relevant packages are correctly installed.

The connection to sql server runs fine whith this code (all teh code in the main function  ...:

`
connection_parameters parameters("odbc", "DSN=VCESara5Db;UID=user;PWD=mypwd");
		parameters.set_option(odbc_option_driver_complete, "0" /* SQL_DRIVER_NOPROMPT */);
		session sql(parameters);
`

Then following an example a found online on the soci web site I ask for data in this way:

rowset<row> rs = (sql.prepare << "select id, description from mytable");

for( rowset<row>::const_iterator it = rs.begin(); it != rs.end(); )
{
    row const& r = *it;

    cout << "id:   " << r.get<int>(0) << '\n'
            << "desc: " << r.get<std::string>(1) << endl;
    ++it;
}
`
and finally close the session

`
sql.close();`
`
The previous code is in a try ... catch block like this:

`
try
{
       ...
}
catch (soci::odbc_soci_error const& e)
{
	cerr << "ODBC Error Code: " << e.odbc_error_code() << endl
		 << "Native Error Code: " << e.native_error_code() << endl
		 << "SOCI Message: " << e.what() << std::endl
		 << "ODBC Message: " << e.odbc_error_message() << endl;
}
catch( exception const & e)
{
	cerr << "DB connection error: " << e.what() << endl;
}
`

On my linux box I correctly installed and setup unixodbc and FreeTDS.

The only output I get in my terminal is:

`
socitest: /usr/include/soci/row.h:68: T soci::row::get(std::size_t) const [with T = int; std::size_t = long unsigned int]: Assertion `holders_.size() >= pos + 1' failed.
`
I googled for a while, but it seems I'm not able to find something relevant to solve this my first issue using soci. Can you please drive me to right source of info, or better, tell me why is this error happening?

Thank you for your work, have a good day.

**Comment by mloskot** [comment]
Look at the [tests](https://github.com/SOCI/soci/tree/master/tests) to see how `rowset` is used there.

p.s. I'd suggest to learn the Markdown basics how to format code snippets in GitHub issues.

**Comment by lucapaganotti** [comment]
You're right, sorry for the bad markdown :-) ... can you please tell me which one is the right example? Otherwise I will search by myself ...
Have a good day.

**Comment by lucapaganotti** [comment]
Ok, I think I found the examples on rowset usage in common_tests.h, sorry for the noise ...

**Comment by lucapaganotti** [comment]
So, I clone github soci master repository, built libraries and then install them on my box. 
I tryed again to mimic the examples given in tests, as you suggested with this trivial code:

```
#include <soci/soci.h>
#include <soci/odbc/soci-odbc.h>
#include <iostream>
#include <iomanip>

using namespace std;
using namespace soci;

int main(int argc, char **argv)
{
	time_t start;
	time_t end;
	int seconds = 0;

	cout << "soci test application" << endl;

	try
	{
		std::string conn_str("DSN=myDsn;Database=myDb;UID=myuser;PWD=mypwd;");
		backend_factory const& backend = *factory_odbc();
		start = time(NULL);

		session sql(backend, conn_str);
		end = time(NULL);
		seconds = end - start;

		cout << "Connected to myDb in " << seconds << " seconds." << endl;

		// mytable has 2 columns and 3 rows
                // first field: int PK
                // second field: varchar(50) not NULL

		int         i = 1;
		int         id;
		std::string desc;

		rowset<row> rs = (sql.prepare << "select * from mytable");

		// iteration through the resultset:
		for( rowset<row>::const_iterator it = rs.begin(); it != rs.end(); ++it, ++i )
		{
		    row const& r = (*it);
		    size_t sz = r.size();
		    cout << "row " << i << " has size " << sz << endl;

		    if( sz != 2 )
		    {
		    	cout << "wrong row size" << endl;
		    }
		    else
		    {
		    	if( r.get_properties(0).get_data_type() != dt_integer )
		    	{
		    		cout << "first field is not an integer" << endl;
		    	}
		    	else
		    	{
		    		if( r.get_indicator(0) == i_ok )
		    		{
		    			id = r.get<int>(0);
		    		}
		    		else
		    		{
		    			cout << "Unable to get id" << endl;
		    		}
		    	}
		    	if( r.get_properties(0).get_data_type() != dt_string )
		    	{
		    		cout << "second field is not a string" << endl;
		    	}
		    	else
		    	{
		    		if( r.get_indicator(1) == i_ok )
		    		{
		    			desc = r.get<std::string>(1);
		    		}
		    		else
		    		{
		    			cout << "Unable to get description" << endl;
		    		}
		    	}

				cout << "id:   " << id << " desc: " << desc << endl;
		    }
		}

		sql.close();
		cout << "Database connection closed." << endl;
	}
	catch (soci::odbc_soci_error const& e)
	{
		cerr << "ODBC Error Code:   " << e.odbc_error_code()    << endl
			 << "Native Error Code: " << e.native_error_code()  << endl
			 << "SOCI Message:      " << e.what()               << endl
			 << "ODBC Message:      " << e.odbc_error_message() << endl;
	}
	catch( exception const & e)
	{
		cerr << "DB connection error: " << e.what() << endl;
	}

	return 0;
}
```
Odbc is correctly configured, the same query via isql against the same database gives me the right results.

This is the output I get after running the above code:

```
soci test application
Connected to  myDb in 0 seconds.
row 1 has size 0
wrong row size
row 2 has size 0
wrong row size
row 3 has size 0
wrong row size
Database connection closed.
```
The rowset has correcly three items, but when I get the i-th row it has no fields ...
I'm not understanding what am I missing ... do I need to further setup something am  missing?

Help is appreciated.

Have a nice evening.
